samlsp/new.go

```diff
@@ -0,0 +1,175 @@
+// Package samlsp provides helpers that can be used to protect web services using SAML.
+package samlsp
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/crewjam/saml"
+	"github.com/crewjam/saml/logger"
+)
+
+// Options represents the parameters for creating a new middleware
+type Options struct {
+	URL               url.URL
+	Key               *rsa.PrivateKey
+	Certificate       *x509.Certificate
+	Intermediates     []*x509.Certificate
+	AllowIDPInitiated bool
+	IDPMetadata       *saml.EntityDescriptor
+	ForceAuthn        bool // TODO(ross): this should be *bool
+
+	// The following fields exist <= 0.3.0, but are superceded by the new
+	// SessionProvider and RequestTracker interfaces.
+	Logger         logger.Interface // DEPRECATED: this field will be removed, instead provide a custom OnError function to handle errors
```

Comment:
I was planning on that, for sure, and semver et al allow breaking changes pre 1.0, but I think it is a little developer unfriendly to make such sweeping changes without a transition / warning. This is based on poking around how people are using the library publicly, although there are breaking changes, keeping these fields around, at least for a while, seems like it would make the breaking changes easier to deal with. 

I'm convincible here, but on balance, my gut for now is to hang on to them for a while.

I'd appreciate any feedback here and/or how strongly people feel about this.